api/all/src/main/java/io/opentelemetry/api/logs/EventBuilder.java

```diff
@@ -0,0 +1,152 @@
+    return put(key, Value.of(value));
+  }
+
+  /** Put the given {@code key} and {@code value} in the payload. */
+  default EventBuilder put(String key, boolean value) {
+    return put(key, Value.of(value));
+  }
+
+  /** Put the given {@code key} and {@code value} in the payload. */
+  default EventBuilder put(String key, String... value) {
+    List<Value<?>> values = new ArrayList<>(value.length);
+    for (String val : value) {
+      values.add(Value.of(val));
+    }
+    return put(key, Value.of(values));
+  }
+
+  /** Put the given {@code key} and {@code value} in the payload. */
+  default EventBuilder put(String key, long... value) {
+    List<Value<?>> values = new ArrayList<>(value.length);
+    for (long val : value) {
+      values.add(Value.of(val));
+    }
+    return put(key, Value.of(values));
+  }
+
+  /** Put the given {@code key} and {@code value} in the payload. */
+  default EventBuilder put(String key, double... value) {
+    List<Value<?>> values = new ArrayList<>(value.length);
+    for (double val : value) {
+      values.add(Value.of(val));
+    }
+    return put(key, Value.of(values));
+  }
+
+  /** Put the given {@code key} and {@code value} in the payload. */
+  default EventBuilder put(String key, boolean... value) {
+    List<Value<?>> values = new ArrayList<>(value.length);
+    for (boolean val : value) {
+      values.add(Value.of(val));
+    }
+    return put(key, Value.of(values));
+  }
+
+  /**
+   * Put the given key and value in the payload.
+   *
+   * <p>NOTE: The key value pair is NOT added to the event attributes. Setting event attributes is
+   * less common than adding entries to the event payload. Use {@link #setAttributes(Attributes)} if
+   * intending the data to be set in attributes instead of the payload.
+   */
+  @SuppressWarnings("unchecked")
+  default <T> EventBuilder put(AttributeKey<T> key, T value) {
+    switch (key.getType()) {
+      case STRING:
+        return put(key.getKey(), (String) value);
+      case BOOLEAN:
+        return put(key.getKey(), (boolean) value);
+      case LONG:
+        return put(key.getKey(), (long) value);
+      case DOUBLE:
+        return put(key.getKey(), (double) value);
+      case STRING_ARRAY:
+        return put(
+            key.getKey(),
+            Value.of(((List<String>) value).stream().map(Value::of).collect(toList())));
+      case BOOLEAN_ARRAY:
+        return put(
+            key.getKey(),
+            Value.of(((List<Boolean>) value).stream().map(Value::of).collect(toList())));
+      case LONG_ARRAY:
+        return put(
+            key.getKey(), Value.of(((List<Long>) value).stream().map(Value::of).collect(toList())));
+      case DOUBLE_ARRAY:
+        return put(
+            key.getKey(),
+            Value.of(((List<Double>) value).stream().map(Value::of).collect(toList())));
+    }
+    return this;
+  }
+
+  /** Put the given {@code key} and {@code value} in the payload. */
+  EventBuilder put(String key, Value<?> value);
```

Comment:
It's a feedback to the existing interface - it seems impossible to add a value without key (e.g. a plain string as a body or an array<Value<T>>) and inconvenient to add a value that's created somewhere else. But it should be easy to address in the future.

Assuming I have some existing POJO for event payload, it'd prefer to do something like 

```java
logger.builder("test-event")
      .put(event.toValue())
      .emit();
```

and now it's not super-convenient:

```java
logger.builder("test-event")
      .put("id", Value.of(event.getId()))
      .put("finishReasons", Value.of(Arrays.stream(event.getFinishReasons()).map(Value::of).toArray(Value[]::new)))
      .put("message", event.getMessage().toValue())
      .emit();
```

Example: https://gist.github.com/lmolkova/1e7aae49483e411a5c7e633490202ee9

Reply:
Do we know of instances where the event payload isn't a map? We can support this, but it seems good to put everything in a map. That way, every piece of data minimally has a key which helps interpret the contents.

Reply:
We might introduce some GenAI events with array (of maps) body.  Another example is someone recording an arbitrary HTTP request/response body as a string or byte array.

Reply:
@jack-berg I was worried that the `put()` methods were confusing whether they applied to attributes or body fields.

what do you think of removing them and just having `setBody()`?

I think the body is a "unit" which you will always want to set in a single place (as opposed to setting fields into it over time), so I'm not sure there is much downside

and I think at some point we want to be able to generate objects from semconv to encapsulate the bodies, which will still fit nicely with a single call to (some kind of) `setBody()` method

Reply:
Initially there was only `setBody(Value)`. I received feedback that the ergonomics of this really suck for users. The `put(String, T)` overloads represented syntactic sugar with the goal of making the user ergonomics reasonably good.

I still think `put(String, T)` overloads do meaningfully improve the experience vs. `put(Value.of(Map.of("key", Value.of("value")))`, and I still think that bodies which are maps are the common case.

What about having both `put(String, T)` overloads, and a `setBody(Value)` escape hatch? We can document that calling `setBody(Value)` erases any previous calls to `put(String, T)`, and calling `put(String, T)` erases any previous calls to `setBody(Value)`, and log warnings if either happens. Essentially, last win semantics with logging if anything gets overridden because its usually an error.

Reply:
> I received feedback that the ergonomics of this really suck for users.

Could the ergonomics be improved by something like this?

```
setBody(ValueBuilder.put(...)
          .put(...)
          .put(...)
          .build()
```

The main reason I don't love the `.put(...)` methods on the `EventBuilder` is because it feels non-obvious whether those are putting body fields or attributes.

Reply:
> Do we know of instances where the event payload isn't a map? We can support this, but it seems good to put everything in a map. That way, every piece of data minimally has a key which helps interpret the contents.

I've been biting my tongue on this for a long time now, because I didn't want to get in the way of progress...but I definitely think it's over-simplifying to force the body to a map. It was very obvious back when it was spelled `AnyValue` because it literally read in English like "Any value....as long as it's a map". 😉 

There are definitely cases where forcing the body content into a fielded map/dictionary causes redundancy. We obviously don't have many events yet in semconv due to the chaos, but one specific example is the `session.end` event ([link](https://github.com/open-telemetry/semantic-conventions/blob/6da7369d013ab6db1f0e68084767713fd05b7a2c/docs/general/session.md#session-end-event)). The semantics could have been that the body contains the string representation of the session id, and that's it. But because of the forced keys/map nature of the body, we've had to cram the session id value into a body field called `session.id`. 

The name of the event should be enough to signify what the content of the body contains, without additional hints or unnecessary bulky/redundant fielding. It would then be up to the event designer to decide how simple or complicated it needs to be.

As another contrived example, let's say a mobile team has built an app, and they want to send an event to indicate a new high score. One semantic approach could be to just define an event that contains the high score value as a long:

```json5
{
  event.name: mygame.highscore,
  body: 1250000 
}
```

and maybe that's sufficient for their uses....while another game might have more detailed requirements:


```json5
{
  event.name: game2.highscore,
  body: {
    score: 1999999,
    level: 17,
    initials: jjp
  }
}
```

I think both are valid uses cases for events. I think that if you always force a map, you'll see things like this:

```json5
{
  event.name: game.highscore,
  body: { 
    value: 123123999
  }
}
```

which IMO has a pretty bad smell. I think we should leave the power/flexibility there and allow the event designers to use the model that suits them best.

Reply:
I'm happy as long as the ergonomics remain good. Adding a ValueBuilder seems like a good approach. Although I _think_ we only need builders for maps and arrays.

Reply:
resolving, I've introduced map and list value builders now